mof_array/pmf/process_mass_data.py

```diff
@@ -94,74 +95,88 @@ def import_simulated_data(mofs_list, sim_results_import, mof_densities, gases):
+        # probability distribution.
+        for mass_exp in all_results_exp:
+            probs_exact = []
+            probs_range = []
+            a, b = 0, float(max(all_masses_sim)) * (1 + mrange)
+            mu, sigma = float(mass_exp), float(std_dev)*float(mass_exp)
+            alpha, beta = ((a-mu)/sigma), ((b-mu)/sigma)
+            for mass_sim in all_results_sim:
+                upper_prob = ss.truncnorm.cdf(float(mass_sim) * (1 + mrange), alpha, beta, loc = mu, scale = sigma)
+                lower_prob = ss.truncnorm.cdf(float(mass_sim) * (1 - mrange), alpha, beta, loc = mu, scale = sigma)
+                probs_range.append(upper_prob - lower_prob)
+                prob_singlepoint = ss.truncnorm(float(mass_sim), alpha, beta, loc=mu, scale=sigma)
+                probs_exact.append(probs_singlepoint)
+            sum_probs_range = sum(probs_range)
+            norm_probs_range = [(i/sum_probs_range) for i in probs_range]
+            sun_probs_exact = sum(probs_exact)
+            norm_probs_exact = [(i/sum_probs_exact) for i in probs_exact]
+
+            # Update dictionary with pmf for each MOF
+            mof_temp_dict = []
             new_temp_dict = []
-            # Calculates all pmfs based on the experimental mass and truncated normal probability distribution.
-            probs = []
-            for mass in all_masses:
-                probs_upper = ss.truncnorm.cdf(float(mass) * (1 + mrange), a, b, loc = my_mean, scale = my_std)
-                probs_lower = ss.truncnorm.cdf(float(mass) * (1 - mrange), a, b, loc = my_mean, scale = my_std)
-                probs.append(probs_upper - probs_lower)
-
-            sum_probs = sum(probs)
-            norm_probs = [(i / sum_probs) for i in probs]
-
-            # Update dictionary with pmf for each MOF, key specified by experimental mass
-            if mof_temp_dict == []:
-                for index in range(len(norm_probs)):
-                    mof_temp_dict = all_results_temp[index].copy()
-                    mof_temp_dict.update({ 'PMF' : norm_probs[index] })
-                    new_temp_dict.extend([mof_temp_dict])
-                mass_temp_dict = new_temp_dict
-            else:
-                for index in range(len(norm_probs)):
-                    mof_temp_dict = mass_temp_dict[index].copy()
-                    mof_temp_dict.update({ 'PMF' : norm_probs[index] })
-                    new_temp_dict.extend([mof_temp_dict])
-                mass_temp_dict = new_temp_dict
-
-        pmf_results.extend(mass_temp_dict)
+            for index in range(len(norm_probs)):
+                mof_temp_dict = all_results_sim[index].copy()
+                mof_temp_dict.update({ 'PMF_Range' : norm_probs_range[index] })
+                mof_temp_dict.update({ 'PMF_Exact' : norm_probs_exact[index] })
+                new_temp_dict.extend([mof_temp_dict])
+            pmf_results.extend(new_temp_dict)
```

Comment:
This code will never run since pmf_results isn't defined. Is this code fixed in later commits? fixes should be squashed in with here.